ambry-store/src/main/java/com.github.ambry.store/LogSegment.java

```diff
@@ -463,11 +489,24 @@ void flush() throws IOException {
 
   /**
    * Closes this log segment
+   * @param shouldSkipDiskFlush whether to skip any disk flush operations.
+   * @throws IOException if there is an I/O error while closing the log segment.
    */
-  void close() throws IOException {
+  void close(boolean shouldSkipDiskFlush) throws IOException {
     if (open.compareAndSet(true, false)) {
-      flush();
-      fileChannel.close();
+      if (!shouldSkipDiskFlush) {
+        flush();
```

Comment:
Why skip this completely, as opposed to catching and ignoring the exception like you do when closing the channel?

Reply:
`flush()` is always attempting to write into disk, which will definitely throw IOException if the disk is bad. The code skips it to avoid unnecessary disk flush operations (note that this method is in `LogSegment`, each store may attempt to write to disk multiple times). However, I am not sure `fileChannel.close()` will invoke IOException if there is disk I/O error. My intention is to give it a try to close the file descriptor of this segment.